language/move-lang/tests/move_check/naming/duplicate_type_parameter_function.exp

```diff
@@ -0,0 +1,33 @@
+error: 
+
+   ┌── tests/move_check/naming/duplicate_type_parameter_function.move:2:9 ───
+   │
+ 2 │     foo<T, T>() {}
+   │            ^ Duplicate type parameter declared with name 'T'
+   ·
+ 2 │     foo<T, T>() {}
+   │         - Previously defined here
+   │
+
+error: 
+
+   ┌── tests/move_check/naming/duplicate_type_parameter_function.move:3:10 ───
+   │
+ 3 │     foo2<T: copyable, T: resource, T>() {}
+   │                       ^ Duplicate type parameter declared with name 'T'
+   ·
+ 3 │     foo2<T: copyable, T: resource, T>() {}
+   │          - Previously defined here
+   │
+
+error: 
+
+   ┌── tests/move_check/naming/duplicate_type_parameter_function.move:3:23 ───
+   │
+ 3 │     foo2<T: copyable, T: resource, T>() {}
+   │                                    ^ Duplicate type parameter declared with name 'T'
+   ·
+ 3 │     foo2<T: copyable, T: resource, T>() {}
```

Comment:
Should the error be pointing at this `T`? Why not the first definition of the `T`?

Reply:
I could.... Currently the unique map updates the loc on the next definition. So the second duplicate points to the first def, and the third points to the second. 

My thoughts was this keeps errors a bit more contextual and close together, but maybe its too confusing? Thoughts?